views/js/payment-components/one-click.js

```diff
@@ -41,22 +40,31 @@ jQuery(function ($) {
 
     var threeDSProcessUrl;
 
-    $('[data-one-click-payment]').each(function (index, element) {
-        var oneClickPaymentConfiguration = $(element).data();
-        var oneClickPaymentMethod = oneClickPaymentConfiguration.oneClickPayment;
-        if (oneClickPaymentMethod) {
-            renderOneClickComponent(oneClickPaymentMethod);
-            fillBrowserInfo();
+    // use this object to iterate through the stored payment methods
+    var checkoutStoredPaymentMethods = window.adyenCheckout.paymentMethodsResponse.storedPaymentMethods;
+
+    // Iterate through the stored payment methods list we got from the adyen checkout component
+    checkoutStoredPaymentMethods.forEach(function (storedPaymentMethod) {
+
+        /*
+            storedPaymentMethod.id = $storedPaymentApiId in the oneclick.tpl
+            if the container doesn't exits don't try to render the component
+        */
+        var storedPaymentMethodContainer = $("#cardContainer-" + storedPaymentMethod.id);
+        if (!storedPaymentMethodContainer.length) {
+            console.log("Something went wrong with the payment method");
```

Comment:
I wouldn't log this here. Logging like this can be seen by shoppers.

Reply:
ok